core/src/main/java/com/linecorp/armeria/client/endpoint/PropertiesEndpointGroup.java

```diff
@@ -141,20 +138,78 @@ public static PropertiesEndpointGroup of(Properties properties, String endpointK
+     * properties whose name starts with {@code endpointKeyPrefix}:
+     *
+     * <pre>{@code
+     * example.hosts.0=example1.com:36462
+     * example.hosts.1=example2.com:36462
+     * example.hosts.2=example3.com:36462
+     * }</pre>
+     *
+     * @param path the path of the file where list of {@link Endpoint}s is loaded from
+     * @param endpointKeyPrefix the property name prefix
+     * @param defaultPort the default port number to use
+     */
+    public static PropertiesEndpointGroup of(Path path, String endpointKeyPrefix, int defaultPort) {
+        validateDefaultPort(defaultPort);
+        return new PropertiesEndpointGroup(requireNonNull(path, "path"),
+                                           requireNonNull(endpointKeyPrefix, "endpointKeyPrefix"),
+                                           defaultPort);
+    }
+
+    /**
+     * Creates a new {@link EndpointGroup} instance that loads the host names (or IP address) and the port
+     * numbers of the {@link Endpoint} from the {@code path} of a resource file. The resource file loads
+     * properties whose name starts with {@code endpointKeyPrefix}:
+     *
+     * <pre>{@code
+     * example.hosts.0=example1.com:36462
+     * example.hosts.1=example2.com:36462
+     * example.hosts.2=example3.com:36462
+     * }</pre>
+     *
+     * @param path the path of the file where list of {@link Endpoint}s is loaded from
+     * @param endpointKeyPrefix the property name prefix
+     */
+    public static PropertiesEndpointGroup of(Path path, String endpointKeyPrefix) {
+        return new PropertiesEndpointGroup(requireNonNull(path, "path"),
+                                           requireNonNull(endpointKeyPrefix, "endpointKeyPrefix"),
+                                           0);
+    }
+
+    private static URL getResourceUrl(ClassLoader classLoader, String resourceName) {
         final URL resourceUrl = classLoader.getResource(resourceName);
         checkArgument(resourceUrl != null, "resource not found: %s", resourceName);
+        return resourceUrl;
+    }
+
+    private static List<Endpoint> loadEndpoints(URL resourceUrl, String endpointKeyPrefix, int defaultPort) {
+
         if (!endpointKeyPrefix.endsWith(".")) {
             endpointKeyPrefix += ".";
         }
+
         try (InputStream in = resourceUrl.openStream()) {
             final Properties props = new Properties();
             props.load(in);
             return loadEndpoints(props, endpointKeyPrefix, defaultPort);
         } catch (IOException e) {
-            throw new IllegalArgumentException("failed to load: " + resourceName, e);
+            throw new IllegalArgumentException("failed to load: " + resourceUrl.getFile(), e);
+        }
+    }
+
+    private static List<Endpoint> loadEndpoints(Path path, String endpointKeyPrefix, int defaultPort) {
+        final URL resourceUrl;
+        try {
+            resourceUrl = path.toUri().toURL();
```

Comment:
We could stop converting a `Path` to a `URL` but just use `Files.newInputStream(Path, ...)`.